tests/Java.Interop.Tools.JavaSource-Tests/SourceJavadocToXmldocGrammar.HtmlBnfTermsTests.cs

```diff
@@ -81,7 +81,38 @@ public void CodeElementDeclaration ()
 			var r = p.Parse ("<code>input.position()</code>");
 			Assert.IsFalse (r.HasErrors (), DumpMessages (r, p));
 			Assert.AreEqual ("<c>input.position()</c>", r.Root.AstNode.ToString ());
-		}
 
+			r = p.Parse ("<code>null</null>");
+			Assert.IsFalse (r.HasErrors (), DumpMessages (r, p));
+			Assert.AreEqual ("<c>null</c>", r.Root.AstNode.ToString ());
+
+			r = p.Parse ("<code>android:label=\"@string/resolve_title\"</code>");
+			Assert.IsFalse (r.HasErrors (), DumpMessages (r, p));
+			Assert.AreEqual ("<c>android:label=\"@string/resolve_title\"</c>", r.Root.AstNode.ToString ());
+
+			r = p.Parse ("<code>Activity.RESULT_OK<code>");
```

Comment:
I would like to extend this test to see what happens with the content after the `<code>`, e.g. from `android/telephony/gsm/SmsManager.java`:

```
     *  The result code will be <code>Activity.RESULT_OK<code> for success,
     *  or one of these errors:
     *  <code>RESULT_ERROR_GENERIC_FAILURE</code>
```

or shorter: `<code>should be code<code> but what about this <code>and this?</code>`.  *Yes*, `<code>should be code<code>` results in `<c>should be code</c>`, but what about the rest?  Would we get:

```
<c>should be code</c><c> but what about this </c><c>and this?</c>
```

which is entirely defensible and reasonable, but is worth "calling out" in the unit tests.

Reply:
We don't appear to create an additional code element for the intermediate content, presumably because we capture the incorrectly placed open <code> tag while parsing the previous code element. I've added another test for this.